### ISSUE TITLE
Unhandled method client/registerCapability

### ISSUE THREAD

**Issue by dwellna** [TypeFox/monaco-languageclient#215]
Hi,

I know there has been the same issue before just a few months back which was closed by the reporter.
But although following his advice and using the latest versions of all related/necessary packages, I still get this response from the language client when the server requests to register capabilities:

{"jsonrpc":"2.0","id":225,"error":{"code":-32601,"message":"Unhandled method client/registerCapability"}}

I tried to debug and it seems like there is no handler registered for that request within vscode-jsonrpc.

This is the request from the server (I am connected to the Omnisharp language server):

{"jsonrpc":"2.0","id":228,"method":"client/registerCapability","params":{"registrations":[{"id":"797c4c87-a2a7-420f-9319-af8a3c7bcea7","method":"textDocument/completion","registerOptions":{"triggerCharacters":["."],"resolveProvider":false,"documentSelector":[{"pattern":"**/*.cs"}]}}]}}

Any suggestions to solve this?

I can add any files needed to this ticket in order to get a solution, but just don't want to overload it right now.

Many thanks!

**Comment by dwellna** [comment]
I tried to dig through the code a bit and it looks like when creating the client I need to call onRequest() to register certain request handlers? (IConnection). I have not yet been able to find any place withing this code base where this happens.
Would this be the editors responsibility? So am I missing a link between the client and the monaco editor?

**Comment by dwellna** [comment]
I got a step further.
Seems like I did a breaking change on the server side when I was figuring out the communication issues I had and then decided to implement my own input and output handlers in the omnisharp language server using Observable<string> and Observer<string> instead of streams.

I just reverted that one particular change and the exchanged message look a lot better.
However after both sides report 'initialized' they basically stop talking to each other.
When typing into the editor or triggering AutoComplete ("No Suggestions") no messages are being sent to the server.

Any ideas?

**Comment by dwellna** [comment]
I got to a point now where server and client successfully initialize and the client is sending regular requests.
Unfortunately in the server, there is no document to synchronize with and therefore also no suggestions for auto complete etc.
I believe the client would need to send an openDocument notification in order for the server to actually create that document and synchronize.
Just need to figure out  how to do this, I noticed the request() method on the client, so that will be my entry point.

I'd very much appreciate any suggestions.

Thanks very much!
